code/modules/research/experimental/computers/research_console.dm

```diff
@@ -0,0 +1,134 @@
+//* This file is explicitly licensed under the MIT license. *//
+//* Copyright (c) 2025 Citadel Station Developers           *//
+
+// TODO: finish this file!
```

Comment:
this file (along with peripherals) are not ticked right now
they'll come in another PR
we're not replacing research just yet with this PR. i just wanted to get the new system in codebase and have working datastructures for things like node / design projection, what is the important part as the new system contains semantics for designs/tech being stored on separate servers from the actual 'mainframe' that hosts the techweb.